aptos-move/framework/supra-framework/sources/automation_registry_state.move

```diff
@@ -0,0 +1,452 @@
+/// Supra Automation Registry State
+///
+/// This contract is part of the Supra Framework and is designed to manage automated task entries
+module supra_framework::automation_registry_state {
+
+    use std::signer;
+    use std::vector;
+    use supra_framework::event;
+
+    use supra_std::enumerable_map::{Self, EnumerableMap};
+
+    use supra_framework::system_addresses;
+    use supra_framework::timestamp;
+    #[test_only]
+    use supra_framework::account;
+
+    friend supra_framework::automation_registry;
+    friend supra_framework::block;
+
+    /// Invalid expiry time: it cannot be earlier than the current time
+    const EINVALID_EXPIRY_TIME: u64 = 1;
+    /// Invalid gas price: it cannot be zero
+    const EINVALID_GAS_PRICE: u64 = 2;
+    /// Invalid max gas amount for automated task: it cannot be zero
+    const EINVALID_MAX_GAS_AMOUNT: u64 = 3;
+    /// Task with provided Id not found
+    const ETASK_NOT_FOUND: u64 = 4;
+    /// Gas amount does not go beyond upper cap limit
+    const EGAS_AMOUNT_UPPER: u64 = 5;
+    /// Automation task not found
+    const EAUTOMATION_TASK_NOT_EXIST: u64 = 6;
+    /// Unauthorized access: the caller is not the owner of the task
+    const EUNAUTHORIZED_TASK_OWNER: u64 = 7;
+    /// Upon new epoch entry failed to propertly calculated committed gas for the next epoch.
+    /// It is greater than current epoch committed gas.
+    const EINVALID_COMMITTED_GAS_CALCULATION: u64 = 8;
+    /// Transactoin hash that registring current task is invalid. Lenght should be 32.
+    /// It is greater than current epoch committed gas.
+    const EINVALID_TXN_HASH: u64 = 9;
+    /// Transactoin hash that registring current task is invalid. Lenght should be 32.
+    /// It is greater than current epoch committed gas.
+    const EINVALID_PAYLOAD: u64 = 10;
+
+    /// The lenght of the transaction hash.
+    const TXN_HASH_LENGTH: u64 = 32;
+
+    /// Conversion factor between microseconds and second
+    const MICROSECS_CONVERSION_FACTOR: u64 = 1_000_000;
+
+    /// It tracks entries both pending and completed, organized by unique indices.
+    struct AutomationRegistryState has key, store {
+        /// A collection of automation task entries that are active state.
+        tasks: EnumerableMap<u64, AutomationTaskMetaData>,
+        current_index: u64,
+        gas_committed_for_next_epoch: u64,
+        automation_gas_limit: u64,
+    }
```

Comment:
at this moment parameters has been split into 2 contexts `automation_registry` and `automation_registry_state` based on their place of utilization. but I think as long as we are going to have more and more configurable parameters it would be better to have dedicated structure for it. But I would suggest to do it in a separate PR e.g in scope of the PR wehn automation-fee related configurations  will be introduced.